src/go/rpk/pkg/cli/cluster/partitions/toggle.go

```diff
@@ -0,0 +1,243 @@
+	cmd.Flags().BoolVarP(&all, "all", "a", false, "If true, enable all partitions")
+
+	cmd.MarkFlagsMutuallyExclusive("partitions", "all")
+	return cmd
+}
+
+func newPartitionDisableCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var (
+		partitions []string
+		all        bool
+	)
+	cmd := &cobra.Command{
+		Use:   "disable [TOPIC]",
+		Short: "Disable partitions of a topic",
+		Long: `Disable partitions of a topic
+
+You may disable all partitions of a topic using the --all flag or you may select 
+a set of topic/partitions to disable with the '--partitions/-p' flag.
+
+The partition flag accepts the format {namespace}/{topic}/[partitions...]
+where namespace and topic are optional parameters. If the namespace is not
+provided, rpk will assume 'kafka'. If the topic is not provided in the flag, rpk
+will use the topic provided as an argument to this command.
+
+DISABLED PARTITIONS
+
+Disabling a partition in Redpanda involves prohibiting any data consumption or
+production to and from it. All internal processes associated with the partition
+are stopped, and it remains unloaded during system startup. This measure aims to
+maintain cluster health by preventing issues caused by specific corrupted
+partitions that may lead to Redpanda crashes. Although the data remains stored
+on disk, Redpanda ceases interaction with the disabled partitions to ensure
+system stability.
+
+EXAMPLES
+
+Disable all partitions in topic 'foo'
+    rpk cluster partitions disable foo --all
+
+Disable partitions 1,2 and 3 of topic 'bar' in the namespace 'internal'
+    rpk cluster partitions disable internal/bar --partitions 1,2,3
+
+Disable partition 1, and 2 of topic 'foo', and partition 5 of topic 'bar' in the 
+'internal' namespace' 
+    rpk cluster partitions disable -p foo/1,2 -p internal/bar/5
+`,
+		Args: cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, topicArg []string) {
+			if !all && partitions == nil {
+				fmt.Println("Please select either specific partitions to disable (--partitions) or select all (--all)")
+				cmd.Help()
+				os.Exit(1)
+			}
+			if all && len(topicArg) == 0 {
+				fmt.Println("You must select a topic")
+				cmd.Help()
+				os.Exit(1)
+			}
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+			out.CheckExitCloudAdmin(p)
+
+			cl, err := adminapi.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			err = runToggle(cmd.Context(), cl, all, topicArg, partitions, "disable")
+			out.MaybeDieErr(err)
+			if all {
+				fmt.Printf("Successfully disabled all partitions in topic %v\n", topicArg[0])
+			} else {
+				fmt.Printf("Successfully disabled the partitions %v\n", partitions)
+			}
+		},
+	}
+	cmd.Flags().StringArrayVarP(&partitions, "partitions", "p", nil, "Comma-separated list of partitions you want to disable. Use --help for additional information")
+	cmd.Flags().BoolVarP(&all, "all", "a", false, "If true, disable all partitions")
```

Comment:
This implies disabling 'all' partitions in the cluster. Whereas `You must select a topic` is thrown during the command execution, but can we make the help text clearer too? e.g. `If true, disable all partitions for specified topic(s)`

Reply:
Is the all flag necessary in these commands?
The reason --all was necessary in `list` is because list _by default_ listed all partitions in the cluster, which could be a lot -- so we want the default to be the help text.

Here, `enable` does nothing if there are no topics specified, and if one topic is specified without flags, I'd expect that all partitions of the topic is enabled (i.e. the topci is wholly enabled) -- so I don't think --all is needed. wdyt?

Reply:
That sounds neat, indeed.

Reply:
w.r.t `--all` flag: @twmb I followed your advice in https://github.com/redpanda-data/redpanda/pull/14909#discussion_r1396529710 and it made sense to me since disabling **all** partitions of a topic mean prohibiting any data consumption or production to and from it... And that sounds dangerous and I'd prefer having that behind a flag. wdyt?